src/solana/index.ts

```diff
@@ -340,50 +376,400 @@ export class Solana extends Blockchain {
+      }),
+    );
+
+    return [createStakeAccountTx, stakeAccountPubkey, []];
+  }
+
+  /** unstake - unstake
+   * @param {string} token - auth API token
+   * @param {string} sender - account blockchain address (staker)
+   * @param {number} lamports - lamport amount
+   * @param {string} source - stake source
+   * @returns {Promise<object>} Promise object with Versioned Tx
+   */
+  public async unstake(
+    token: string,
+    sender: string,
+    lamports: number,
+    source: string,
+  ): Promise<ApiResponse<VersionedTransaction>> {
+    // Check if the token is valid
+    const isTokenValid = await CheckToken(token);
+    if (!isTokenValid) {
+      this.throwError('INVALID_TOKEN_ERROR');
+    }
+
+    try {
+      const delegations = await this.getDelegations(sender);
+
+      const stakeAccounts = delegations.result.map((delegationAcc) => {
+        return {
+          pubkey: delegationAcc.pubkey,
+          account: parsedAccountInfoToStakeAccount(delegationAcc.account),
+        };
+      });
+
+      const epochInfo = await this.connection.getEpochInfo();
+      const tm = this.timestampInSec();
+
+      let totalActiveStake = new BigNumber(0);
+      const activeStakeAccounts = stakeAccounts.filter((acc) => {
+        const isActive = !(
+          isLockupInForce(acc.account.data.info.meta, epochInfo.epoch, tm) ||
+          stakeAccountState(acc.account.data, epochInfo.epoch) !==
+            StakeState.Active
+        );
+        if (isActive && acc.account.data.info.stake !== null) {
+          totalActiveStake = totalActiveStake.plus(
+            acc.account.data.info.stake.delegation.stake,
+          );
+        }
+
+        return isActive;
+      });
+
+      let lamportsBN = new BigNumber(lamports);
+      if (totalActiveStake.lt(lamportsBN))
+        throw this.throwError('NOT_ENOUGH_ACTIVE_STAKE_ERROR');
+
+      // Desc sorting
+      activeStakeAccounts.sort((a, b): number => {
+        if (
+          a.account.data.info.stake === null ||
+          b.account.data.info.stake === null
+        ) {
+          return 0;
+        }
+
+        if (
+          a.account.data.info.stake.delegation.stake.lte(
+            b.account.data.info.stake.delegation.stake,
+          )
+        ) {
+          return 1;
+        }
+
+        return -1;
+      });
+
+      const accountsToDeactivate = [];
+      const accountsToSplit = [];
+      let i = 0;
+      while (
+        lamportsBN.gt(new BigNumber(0)) &&
+        i < activeStakeAccounts.length
+      ) {
+        const lBN = new BigNumber(lamports);
+        const acc = activeStakeAccounts[i];
+        if (acc === undefined || acc.account.data.info.stake === null) {
+          i++;
+          continue;
+        }
+        const stakeAmount = acc.account.data.info.stake.delegation.stake;
+
+        // If reminder amount less than min stake amount stake account automatically become disabled
+        if (
+          stakeAmount.comparedTo(lBN) <= 0 ||
```

Comment:
maybe separate const for this 
```suggestion
          const isBelowThreshold =  stakeAmount.lte(lamportsBN) || stakeAmount.minus(lamportsBN).lt(MIN_AMOUNT);
```